app/lib/dns-server.ts

```diff
@@ -0,0 +1,239 @@
+      }
+    });
+  } catch (error) {
+    throw new Error(`Error while deleting hosted zone: ${error}`);
+  }
+};
+
+export const createRecord = async (
+  type: RecordType,
+  name: string,
+  value: string,
+  hostedZoneId: string
+): Promise<string> => {
+  try {
+    const command = new ChangeResourceRecordSetsCommand({
+      ChangeBatch: {
+        Changes: [
+          {
+            Action: "CREATE",
+            ResourceRecordSet: {
+              Name: name,
+              Type: type,
+              ResourceRecords: [
+                {
+                  Value: value,
+                },
+              ],
+            },
+          },
+        ],
+      },
+      HostedZoneId: hostedZoneId,
+    });
+    const response = await route53Client.send(command);
+
+    if (!response.ChangeInfo?.Id) {
+      throw new Error(`Could not create a record`);
+    } else {
+      return response.ChangeInfo?.Id;
+    }
+  } catch (error) {
+    throw new Error(
+      `Something went wrong while creating resource record: ${error}`
+    );
+  }
+};
+
+export const updateRecord = async (
+  type: RecordType,
+  name: string,
+  value: string,
+  hostedZoneId: string
+): Promise<string> => {
+  try {
+    const command = new ChangeResourceRecordSetsCommand({
+      ChangeBatch: {
+        Changes: [
+          {
+            Action: "UPSERT",
+            ResourceRecordSet: {
+              Name: name,
+              Type: type,
+              ResourceRecords: [
+                {
+                  Value: value,
+                },
+              ],
+            },
+          },
+        ],
+      },
+      HostedZoneId: hostedZoneId,
+    });
+    const response = await route53Client.send(command);
+
+    if (!response.ChangeInfo?.Id) {
+      throw new Error(`Could not update the record`);
+    } else {
+      return response.ChangeInfo?.Id;
+    }
+  } catch (error) {
+    throw new Error(
+      `Something went wrong while updating resource record: ${error}`
+    );
+  }
+};
+
+export const deleteRecord = async (
+  type: RecordType,
+  name: string,
+  value: string,
+  hostedZoneId: string
+): Promise<string> => {
+  try {
+    const command = new ChangeResourceRecordSetsCommand({
+      ChangeBatch: {
+        Changes: [
+          {
+            Action: "DELETE",
+            ResourceRecordSet: {
+              Name: name,
+              Type: type,
+              ResourceRecords: [
+                {
+                  Value: value,
+                },
+              ],
+            },
+          },
+        ],
+      },
+      HostedZoneId: hostedZoneId,
+    });
+
+    const response = await route53Client.send(command);
+
+    if (!response.ChangeInfo?.Id) {
+      throw new Error("Could not delete the record");
+    } else {
+      return response.ChangeInfo?.Id;
+    }
+  } catch (error) {
+    throw new Error(
+      `Something went wrong while deleting resource record: ${error}`
+    );
+  }
+};
+
+export const listRecords = async (
```

Comment:
I don't think we need this.  Our database will be our query source.

Reply:
I see. I will remove it.